element_array_ephys/plotting/unit_level.py

```diff
@@ -0,0 +1,197 @@
+        template="simple_white",
+        width=350,
+        height=350,
+    )
+    return fig
+
+
+def plot_correlogram(
+    spike_times: np.ndarray, bin_size: float = 0.001, window_size: int = 1
+) -> go.Figure:
+
+    from brainbox.singlecell import acorr
+
+    correlogram = acorr(
+        spike_times=spike_times, bin_size=bin_size, window_size=window_size
+    )
+    df = pd.DataFrame(
+        data={"correlogram": correlogram},
+        index=pd.RangeIndex(
+            start=-(window_size * 1e3) / 2,
+            stop=(window_size * 1e3) / 2 + bin_size * 1e3,
+            step=bin_size * 1e3,
+        ),
+    )
+    df["lags"] = df.index  # in ms
+
+    fig = go.Figure()
+    fig.add_trace(
+        go.Scatter(
+            x=df["lags"],
+            y=df["correlogram"],
+            mode="lines",
+            line=dict(color="black", width=1),
+            hovertemplate="%{y}<br>" + "%{x:.2f} ms<extra></extra>",
+        )
+    )
+    fig.update_layout(
+        title="Auto Correlogram",
+        xaxis_title="Lags (ms)",
+        yaxis_title="Count",
+        template="simple_white",
+        width=350,
+        height=350,
+        yaxis_range=[0, None]
+    )
+    return fig
+
+
+def plot_depth_waveforms(
+    unit_key: dict,
+    y_range: float = 60,
+) -> go.Figure:
+
+    from .. import probe
+    from .. import ephys_no_curation as ephys
+
+    sampling_rate = (ephys.EphysRecording & unit_key).fetch1(
+        "sampling_rate"
+    ) / 1e3  # in kHz
+
+    probe_type, peak_electrode = (ephys.CuratedClustering.Unit & unit_key).fetch1(
+        "probe_type", "electrode"
+    )  # electrode where the peak waveform was found
+
+    electrodes, coord_y = (
+        probe.ProbeType.Electrode & f"probe_type='{probe_type}'"
+    ).fetch("electrode", "y_coord")
+
+    peak_coord_y = coord_y[electrodes == peak_electrode][0]
+
+    coord_ylim_low = (
+        coord_y.min()
+        if (peak_coord_y - y_range) <= coord_y.min()
+        else peak_coord_y - y_range
+    )
+    coord_ylim_high = (
+        coord_y.max()
+        if (peak_coord_y + y_range) >= coord_y.max()
+        else peak_coord_y + y_range
+    )
+
+    tbl = (
+        (probe.ProbeType.Electrode)
+        & f"probe_type = '{probe_type}'"
+        & f"y_coord BETWEEN {coord_ylim_low} AND {coord_ylim_high}"
+    )
+    electrodes_to_plot = tbl.fetch("electrode")
+
+    coords = np.array(tbl.fetch("x_coord", "y_coord")).T  # x, y coordinates
+
+    waveforms = (
+        ephys.WaveformSet.Waveform
+        & unit_key
+        & f"electrode IN {tuple(electrodes_to_plot)}"
+    ).fetch("waveform_mean")
+    waveforms = np.stack(waveforms)  # all mean waveforms of a given neuron
+
+    x_min, x_max = np.min(coords[:, 0]), np.max(coords[:, 0])
+    y_min, y_max = np.min(coords[:, 1]), np.max(coords[:, 1])
+
+    # Spacing between channels (in um)
+    x_inc = np.abs(np.diff(coords[:, 0])).min()
+    y_inc = np.unique((np.abs(np.diff(coords[:, 1])))).max()
```

Comment:
This calculations do not appear to be generally correct or robust. Need a better algorithm.